### ISSUE TITLE
Clickable links have not to include a punctuation mark if it is the last character in the link

### ISSUE THREAD

**Issue by vrubleg** [notepad-plus-plus/notepad-plus-plus#3353]
Currently, Notepad++ includes punctuation marks into urls, but it is not right in 99% of cases. Usually, punctuation marks has to be excluded if they are the last symbol in the URL.

List of characters which have to be excluded from the URL if they are the last characters:
```
.,:;?!()[]{}
```

**Comment by vrubleg** [comment]
It has to ignore last chars if all of them are from the list. For example, (http://example.com/), has to skip last ) and , symbols.

**Comment by GitLaboratory** [comment]
I agree, it's worth fixing.

**Comment by richlux** [comment]
Yes, this issue is annoying. It makes URLs in notepad++ almost useless for me because they often have punctuation at the end.

**Comment by irgendsontyp** [comment]
@vrubleg, according to your your comment, "http://en.wikipedia.org/wiki/Saw_(film)" wouldn't be a valid url then? Where did you find the information that the characters you mentioned are not allowed at the end of an url/uri?

**Comment by richlux** [comment]
I'm sure it's not as simple as just skipping punctuation at the end, but somehow it should know where the URL starts and ends. Many other programs do it correctly, such as Textpad, Outlook, etc.

Rich

**Comment by vrubleg** [comment]
In can count opening and closing brackets for catching such situations. But, actually, ( and ) are pretty rare in the URLs. The problem when some unnecessary characters are included into an URL is much more frequent now. I hope, that it will be fixed.

**Comment by botmtl** [comment]
With the popularity of markdown on the rise, this went from a minor annoyance that would happen only once in a while to very annoying.  With markdown links being coded the way they are, i.e. the text being inclosed in brakets and the links enclosed in parenthesis.

**Comment by vrubleg** [comment]
@botmtl, all other modern editors are able to handle it. Probably, it worth to switch to Visual Studio Code. It even has Notepad++ keymap plugin from Microsoft itself, so you can switch easily. The only issue here is "Chrome-like" rendering of fonts. For some reason, Chrome always had ugly font smoothing and it is true for Electron-based applications also. But if you are used to Chrome as a browser, I believe you'll be OK about it.

**Comment by vrubleg** [comment]
And yeah, if the authors of Notepad++ decide to implement this nice feature, it worth to add that if there is a quote (', " or ` and a lot of others if you want to support Unicode quotes also) before the link, the same quote after the link should be ignored.

**Comment by Moorelife** [comment]
By the way, a quick workaround for a document that has many URLs like (http://x.y), is to do a global replace  of ')' by ' )', effecitively cutting the underlining before the ).

**Comment by Phroneris** [comment]
I want this!

**Comment by vrubleg** [comment]
@alef162 the new parser can't handle this properly:

```[url=https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-adddlldirectory]AddDllDirectory[/url]```

It is expected that the URL parsing should be stopped at first `]`.

Another example:

```[https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-adddlldirectory]AddDllDirectory]```
It sould stop at first `]`, bit it goes till second `]`.

**Comment by vkleinschmidt** [comment]
This is still an issue in 2023 in Notepad++ v8.5.3. URLs in code are typically quoted strings. The opening quote is clearly not part of the URL, so naturally the closing quote isn't either, but Notepad++ still thinks it is.
